### PR TITLE
Fix bug

### LINKED ISSUE #140: Command recognition not working properly
### Exaples

``` bash
minishell$ $$
minishell: $: command not found
```

### LINKED ISSUE #211: Fix expand variable test case
- [ ] echo "$ USER"
```
bash5.2$ echo "$ USER"
$ USER

minishell$ echo "$ USER"
 UESR
```

### PR DESCRIPTION
**Pull request by alex216**
fix #211 
fix #140

## Summary by CodeRabbit

- **New Features**
	- Enhanced variable expansion logic to handle whitespace characters following a variable.
	- Improved handling of quoted strings in variable processing.

- **Bug Fixes**
	- Improved handling of dollar sign followed by space and tab in variable expansion.

- **Tests**
	- Added new test cases for variable expansion with space and tab following a dollar sign. 
	- Introduced tests for handling quoted strings in variable expansion.

### CODE REVIEW

File: src/expansion/trim_expanded_variable.c [hunk 1]
```diff
@@ -35,7 +35,7 @@ char	*trim_expanded_variable(t_string *expand_info, t_ctx *ctx)
 	char	*result;
 
 	consume_char(expand_info);
-	if (*expand_info->right == '\0')
+	if (*expand_info->right == '\0' || ft_strchr(" \t", *expand_info->right))
```
Comment (rask24) [inline src/expansion/trim_expanded_variable.c:38]: The condition `*expanded_info->right == '\0'` fails to handle quoted variables correctly.
For example `echo "$HOME$"`

We could also consider preserving '$' when followed by non-identifier characters (e.g., `echo $/` should display '$/')

``` bash
bash-5.2$ echo $/
$?
bash-5.2$ echo $:
$:
```

**Comment by alex216** [comment]
@rask24 please check the commits.

**Review by rask24 — APPROVED** [review]
@LGTM 🚆

**Comment by rask24** [comment]
fix #140

**Pull request merged**